dashboard/app/models/user.rb

```diff
@@ -899,7 +907,14 @@ def User.handle_async_op(op)
     raise 'Model must be User' if op['model'] != 'User'
     case op['action']
       when 'track_level_progress'
-        User.track_level_progress_sync(op['user_id'], op['level_id'], op['script_id'], op['new_result'], op['submitted'])
+        User.track_level_progress_sync(
```

Comment:
ha, with the named params you could almost just call `User.track_level_progress_sync(op)`